drafts/web-editor-monaco-with-language-server-support.md

```diff
@@ -0,0 +1,971 @@
+      [/\d+/, 'number'],
+
+      // delimiter: after number because of .\d floats
+      [/[;,.]/, 'delimiter'],
+
+      // strings
+      [/"([^"\\]|\\.)*$/, 'string.invalid' ],  // non-teminated string
+      [/"/,  { token: 'string.quote', bracket: '@open', next: '@string' } ],
+
+      // characters
+      [/'[^\\']'/, 'string'],
+      // [/(')(@escapes)(')/, ['string','string.escape','string']],
+      [/'/, 'string.invalid']
+    ],
+
+    comment: [
+      [/[^\/*]+/, 'comment' ],
+      [/\/\*/,    'comment', '@push' ],    // nested comment
+      ["\\*/",    'comment', '@pop'  ],
+      [/[\/*]/,   'comment' ]
+    ],
+
+    string: [
+      [/[^\\"]+/,  'string'],
+      // [/@escapes/, 'string.escape'],
+      [/\\./,      'string.escape.invalid'],
+      [/"/,        { token: 'string.quote', bracket: '@close', next: '@pop' } ]
+    ],
+
+    whitespace: [
+      [/[ \t\r\n]+/, 'white'],
+      [/\/\*/,       'comment', '@comment' ],
+      [/\/\/.*$/,    'comment'],
+    ],
+  },
+};
+```
+
+Then we need to update the `languageDef` section in the `ls.config.ts` file by setting the `syntaxDefinitions` into the `monarchLanguage` property as shown below:
+```typescript
+  languageDef: {
+    languageExtensionConfig: {
+      id: LANG_ID, 
+      extensions: [LANG_EXTENSION],
+    },
+    monarchLanguage: syntaxDefinitions,
+  },
+```
+
+After adding the language syntax definitions, the final configuration file will look like the example below:
+
+_ls.config.ts_
+```typescript
+import { UserConfig } from 'monaco-editor-wrapper';
+
+export const LANG_ID = 'hello';
+export const LANG_EXTENSION = 'hello';
+
+const LS_WS_URL = 'ws://localhost:8080/ls';
+
+export const lsConfig: UserConfig = {
+  wrapperConfig: {
+    editorAppConfig: {
+      $type: 'classic',
+      codeResources: {
+        main: {
+          text: '',
+          fileExt: LANG_EXTENSION,
+        },
+      },
+      useDiffEditor: false,
+      languageDef: {
+        languageExtensionConfig: {
+          id: LANG_ID,
+          extensions: [LANG_EXTENSION],
+        },
+        monarchLanguage: syntaxDefinitions,
+      },
+    },
+  },
+  languageClientConfig: {
+    languageId: LANG_ID,
+    options: {
+      $type: 'WebSocketUrl',
+      url: LS_WS_URL,
+      startOptions: {
+        onCall: () => {
+          console.log('Connected to socket.');
+        },
+        reportStatus: true,
+      },
+      stopOptions: {
+        onCall: () => {
+          console.log('Disconnected from socket.');
+        },
+        reportStatus: true,
+      },
+    },
+  },
+};
+```
+
+### Step 5: Starting up the language server and web editor
+
+Starting the language server
+
+```shell
+java -jar target/hellols-0.0.1-SNAPSHOT.jar
+```
+
+Starting the Angular application
+
+```shell
+ng serve
+```
+
+below is the final look.
+
+![web-editor](https://cdn.hashnode.com/res/hashnode/image/upload/v1717756040911/IFeF6apmJ.png?auto=format)
+
+## Creating Web Editor using React
```

Comment:
As this article is already quite a long read you could ask yourself if you want to split it to two (lsp & angular / react) or even three articles (lsp / angular / react)

Reply:
i like this idea, in hashnode we can do series